inflation/dataset/parse.py

```diff
@@ -0,0 +1,410 @@
+import json
+import datetime
+import logging
+import _pickle as cPickle
+import os
+import tempfile
+
+import pandas as pd
+import gzip
+
+from dataclasses import dataclass
+from pathlib import Path
+from bs4 import BeautifulSoup
+
+from common.customized_logging import configure_logging
+from common.utils.mathematical_conversation_utils import convert_price_to_us
+
+
+OUTPUT_PATH_DIR = Path(__file__).parents[2] / "inflation-resources/data/"
+PARSER_OUTPUT_DIR = f"{OUTPUT_PATH_DIR}/parser"
+
+NOT_AVAILABLE = "___NA___"
+
+configure_logging()
+logger = logging.getLogger(__name__)
+
+
+@dataclass
+class InflationDataRecord:
+    item_code: str
+    item_name: str
+    source: str
+    product_name: str
+    product_url: str
+    product_code: str  # item identifier for this data.
+    product_brand: str
+    price: float
+    currency: str
+    in_stock: bool
+    sample_date: datetime.datetime
+
+
+class InflationDataset:
+    def __init__(self, data: list):
+        self.dataset = data
+
+    def __len__(self):
+        return len(self.dataset)
+
+    def __iter__(self):
+        return InflationDatasetIterator(self.dataset)
+
+    def __getitem__(self, s):
+        if s < 0:
+            s = self.__len__() - s
+            if s < 0 or s >= self.__len__():
+                raise IndexError
+            else:
+                return self.dataset[s]
+        else:
+            start, stop, step = s.indices(self.__len__())
+            indxs = list(range(start, stop, step))
+            return [self.dataset[i] for i in indxs]
+
+    def save(self, output_file_name):
+        """
+        It saves dataset records by serializing.
+        """
+
+        with open(output_file_name, "wb") as f:
+            cPickle.dump(self, f)
+        return output_file_name
+
+    @classmethod
+    def read(cls, file_path):
+        with open(file_path, "rb") as input_file:
+            return cPickle.load(input_file)
+
+    def to_df(self):
+        return pd.DataFrame([record.__dict__ for record in self.dataset])
+
+
+class InflationDatasetIterator:
+    def __init__(self, inflation_dataset):
+        self.dataset = inflation_dataset
+        self._i = 0
+
+    def __iter__(self):
+        return self
+
+    def __next__(self):
+        if self._i >= len(self.dataset):
+            raise StopIteration
+        else:
+            result = self.dataset[self._i]
+            self._i += 1
+            return result
+
+
+class ParserManager:
+    def __init__(self, parsers: dict, crawler_client=None):
+        self.parsers = parsers
+        self.crawler_client = crawler_client
```

Comment:
let's remove all the client logic elsewhere (`main.py`). This class requires only a file.